src/types/Course/Preference.ts

```diff
@@ -23,8 +25,25 @@ export const PreferenceQuery = extendType({
   definition(t) {
     t.nonNull.field('survey', {
       type: TeachingPreferenceSurvey,
-      description: 'Get Teaching Preference Survey',
-      resolve: () => ({ courses: [] }),
+      description: 'Get Teaching Preference Survey for the current user',
+      resolve: async (_, __, ctx) => {
```

Comment:
```suggestion
      resolve: async (_, __, { prisma, token } ) => {
```
Super small nit but I think you can just destructure here and use `prisma` and `token` directly without having to call `ctx`